src/main/java/faang/school/projectservice/validator/CreateSubProjectDtoValidatorImpl.java

```diff
@@ -0,0 +1,65 @@
+package faang.school.projectservice.validator;
+
+import faang.school.projectservice.dto.project.CreateSubProjectDto;
+import faang.school.projectservice.dto.project.ProjectDto;
+import faang.school.projectservice.dto.project.UpdateSubProjectDto;
+import faang.school.projectservice.model.Project;
+import faang.school.projectservice.model.ProjectVisibility;
+import faang.school.projectservice.repository.ProjectRepository;
+import jakarta.validation.ValidationException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+@RequiredArgsConstructor
+public class CreateSubProjectDtoValidatorImpl implements CreateSubProjectDtoValidator {
+    private final ProjectRepository projectRepository;
+
+    @Override
+    public void validateOnCreate(CreateSubProjectDto projectDto) {
+        validateVisibility(projectDto);
+        validateIsParentProjectExist(projectDto);
+    }
+    @Override
+    public void validateOnUpdate(UpdateSubProjectDto projectDto) {
+        validateIsProjectExist(projectDto);
+        validateIsSubProjectsHaveSameStatus(projectDto);
+    }
+
+    private void validateVisibility(CreateSubProjectDto projectDto) {
+        Project project = projectRepository.getProjectById(projectDto.getParentId());
+        if (project.getVisibility().equals(ProjectVisibility.PUBLIC)
+                && projectDto.getVisibility().equals(ProjectVisibility.PRIVATE)) {
+            throw new ValidationException("Project visibility is public");
+        }
+    }
+
+    private void validateIsParentProjectExist(CreateSubProjectDto projectDto) {
+        if (projectRepository.existsById(projectDto.getParentId())) {
+            return;
+        } else throw new ValidationException("Parent project does not exist");
```

Comment:
if (!projectRepository.existsById(projectDto.getParentId())) {
           throw new ValidationException("Parent project does not exist");
}